server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -562,7 +637,18 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(name, timeStampField, indices, generation, metadata, hidden, replicated, allowCustomRouting);
+        return Objects.hash(name, type, timeStampField, indices, generation, metadata, hidden, replicated, allowCustomRouting);
+    }
+
+    public enum Type {
```

Comment:
Can we reuse IndexMode here? If this is not possible or impractical, I think we should at least reuse the same names here and in IndexMode.

Reply:
> Can we reuse IndexMode here?

I will look into this as well and conjunction with Nik's comment about not have to use a type at all.

> I think we should at least reuse the same names here and in IndexMode.

Agreed, at least the naming should be consistent between the two enums.